### PR TITLE
Coverity out-of-bounds access error

### PR DESCRIPTION
**Pull request by krader1961**
Incorrect loop constructs in `check_limits()` causes dereference past
the end of the `fltconst` and `intconst` arrays.

Coverity CID#253767

### CODE REVIEW

**Comment by krader1961** [comment]
I pushed again to rebase on top of the `master` branch.

File: src/cmd/ksh93/sh/arith.c [hunk 1]
```diff
@@ -295,71 +304,79 @@ static const Fltconst_t fltconst[] = {
...
+static_fn Namval_t *check_limits(Shell_t *shp, char *cp) {
+    char *ep = strrchr(cp, '.');
+    if (!ep) return NULL;  // no attribute name
+
+    *ep = 0;
+    Namval_t *np = nv_open(cp, shp->var_tree, NV_VARNAME | NV_NOADD | NV_NOFAIL);
+    *ep++ = '.';
+    if (!np || !nv_isattr(np, NV_INTEGER)) return NULL;  // var type not integer or float
+
+    if (nv_isattr(np, NV_DOUBLE) == NV_DOUBLE) {
```
Comment (siteshwar) [inline src/cmd/ksh93/sh/arith.c]: IIRC you questioned this pattern, why are you using it ?

Comment (krader1961) [reply src/cmd/ksh93/sh/arith.c]: Yes I did. See issue #855. But changing it is outside the scope of this fix. And omitting the `== NV_DOUBLE` is only safe if you know the symbol has a single bit set.

Comment (krader1961) [reply src/cmd/ksh93/sh/arith.c]: Which is precisely why I want to change the definition of `nv_isattr()`. Notice that the preceding  use omits the `== NV_INTEGER` test. The problem here is that this is the definition for `NV_DOUBLE`:

```
#define NV_DOUBLE (NV_INTEGER | NV_ZFILL)
```

This is without a doubt one of the worst APIs I've seen I've seen in this project.

**Pull request closed without merging**